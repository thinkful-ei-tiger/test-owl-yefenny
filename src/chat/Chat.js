import React, { Component } from 'react';
import chatEvents from '../chatEvents';
import Event from '../event/Event';
import './Chat.css';

export default class Chat extends Component {
  render() {
    const events = chatEvents
      .sort((a, b) => b.timestamp < a.timestamp)
      .map((event, i) => {
        return <Event key={i} event={event} />;
      });
    return <div>{events}</div>;
  }
}
