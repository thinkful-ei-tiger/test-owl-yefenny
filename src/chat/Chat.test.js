import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Chat from './Chat';

describe('Chat component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Chat />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('UI renders without problem', () => {
    const tree = renderer.create(<Chat />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
