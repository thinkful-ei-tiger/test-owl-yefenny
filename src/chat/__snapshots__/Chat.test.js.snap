// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component UI renders without problem 1`] = `
<div>
  <div
    className="event"
  >
    <div
      className="message-type"
    >
      <div
        className="message-avatar"
      >
        <img
          alt="profile"
          src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
        />
      </div>
      <div
        className="message-information"
      >
        <div
          className="message-header"
        >
          <h2>
            Axe Kubicka
          </h2>
          <span>
            07:50 a.m.
          </span>
        </div>
        <div
          className="message"
        >
          <p>
            Hello world
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="event"
  >
    <div>
      <span>
        <strong>
          Caty Flucker
        </strong>
         gave a thumbs up üëç
      </span>
    </div>
  </div>
  <div
    className="event"
  >
    <div>
      <span>
        <strong>
          Frank Runciman
        </strong>
         gave a thumbs down üëé
      </span>
    </div>
  </div>
  <div
    className="event"
  >
    <div>
      <span>
        <strong>
          Koren Templeton
        </strong>
         raise their hand ‚úã
      </span>
    </div>
  </div>
  <div
    className="event"
  >
    <div>
      <span>
        <strong>
          Koren Templeton
        </strong>
         clapped üëè
      </span>
    </div>
  </div>
  <div
    className="event"
  >
    <div>
      <span>
        <strong>
          Koren Templeton
        </strong>
         joined
      </span>
    </div>
  </div>
  <div
    className="event"
  >
    <div>
      <span>
        <strong>
          Ashla Attwool
        </strong>
         left
      </span>
    </div>
  </div>
  <div
    className="event"
  >
    <div>
      <span>
        <strong>
          Axe Kubicka
        </strong>
         joined the stage
      </span>
    </div>
  </div>
  <div
    className="event"
  >
    <div>
      <span>
        <strong>
          Axe Kubicka
        </strong>
         left the stage
      </span>
    </div>
  </div>
</div>
`;
